internal/store/engine_state.go

```diff
@@ -111,6 +114,51 @@ func (ms *ManifestState) MostRecentPod() Pod {
 	return ms.PodSet.MostRecentPod()
 }
 
+func (ms *ManifestState) NextBuildReason() BuildReason {
+	reason := BuildReasonNone
+	if len(ms.PendingFileChanges) > 0 {
+		reason = reason.With(BuildReasonFlagFiles)
+	}
+	if !ms.PendingManifestChange.IsZero() {
+		reason = reason.With(BuildReasonFlagConfig)
+	}
+	if !ms.StartedFirstBuild {
+		reason = reason.With(BuildReasonFlagInit)
+	}
+	if !ms.CrashRebuildInProg {
+		reason = reason.With(BuildReasonFlagCrash)
+	}
+	return reason
+}
+
+// Whether a change at the current time would trigger a build.
+func (ms *ManifestState) IsPendingTime(t time.Time) bool {
```

Comment:
these function names aren't immediately apparent to me, fwiw

Reply:
fair, but i'm not sure i have an idea for a better one. i added comments.

Reply:
nit: a change at the GIVEN time ("current time" implies time.Now)

Reply:
great minds!
https://github.com/windmilleng/tilt/blob/master/internal/store/engine_state.go#L134